module/os/freebsd/zfs/zfs_vnops_os.c

```diff
@@ -1735,7 +1735,7 @@ zfs_readdir(vnode_t *vp, zfs_uio_t *uio, cred_t *cr, int *eofp,
 	/*
 	 * Quit if directory has been removed (posix)
 	 */
-	if ((*eofp = zp->z_unlinked) != 0) {
+	if ((*eofp = (zp->z_unlinked != 0)) != 0) {
```

Comment:
I am not sure what this supposed to fix.  `z_unlinked` is boolean_t, and while it does not enforce values unlike bool, and while there is some mess with it sometimes assigned 0, I don't see any place where it would be assigned anything non-boolean.

Reply:
rmacklem@f.o reported that he got panic due to *eofp == -1. See https://lists.freebsd.org/archives/freebsd-current/2025-August/008428.html

Even if it is fixed more in zfs, I still think that this change is right since we are crossing interface boundary there, and it is better to be explicit.